src/libs/fileDownload/index.android.js

```diff
@@ -0,0 +1,83 @@
+import {PermissionsAndroid} from 'react-native';
+import RNFetchBlob from 'rn-fetch-blob';
+import * as FileUtils from './FileUtils';
+
+/**
+ * Android permission check to store images
+ * @returns {Promise<Boolean>}
+ */
+function hasAndroidPermission() {
+    // Read and write permission
+    const writePromise = PermissionsAndroid.check(PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE);
+    const readPromise = PermissionsAndroid.check(PermissionsAndroid.PERMISSIONS.READ_EXTERNAL_STORAGE);
+
+    return Promise.all([writePromise, readPromise]).then(([hasWritePermission, hasReadPermission]) => {
+        if (hasWritePermission && hasReadPermission) {
+            return true; // Return true if permission is already given
+        }
+
+        // Ask for permission if not given
+        return PermissionsAndroid.requestMultiple([
+            PermissionsAndroid.PERMISSIONS.READ_EXTERNAL_STORAGE,
+            PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE,
+        ]).then(status => status['android.permission.READ_EXTERNAL_STORAGE'] === 'granted'
+                    && status['android.permission.WRITE_EXTERNAL_STORAGE'] === 'granted');
+    });
+}
```

Comment:
What if the user has given read permission but not write permission?  Does it matter if we request it again?

Reply:
I haven't tested this. Also considering that this issue was specifically for iOS, in Android I just checked if it doesn't break anything. Do you want me to test this thoroughly for Android as well?

Reply:
During my testing, Android only ask once for permission which means it combines them together. So if you deny it showns and error. Downloading the file again, asks for permission.